### PR TITLE
feat(PSP-1219): add provider for azure agentless integration

### PR DESCRIPTION
**Pull request by qianxuweiren**
***Issue***: https://lacework.atlassian.net/browse/PSP-1219

***Description:***
This adds lacework integration provider for Azure Agentless support.

Added an integration test for the change.

`cd /Users/aoz/dev/terraform-provider-lacework/ && make build && make install && go mod vendor && cd integration && go test -v -run "TestIntegrationAzureAgentlessScanningCreate"` passed

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,6 +11,8 @@ require (
 	golang.org/x/text v0.13.0
 )
 
+replace	github.com/lacework/go-sdk v1.37.0 => /Users/aoz/dev/go-sdk
```
Comment (qianxuweiren) [inline go.mod:14]: was added for local test override. Will remove

File: examples/resource_lacework_integration_azure_agentless_scanning/main.tf [hunk 2]
```diff
@@ -0,0 +1,116 @@
+terraform {
+  required_providers {
+    lacework = {
+      source = "lacework/lacework"
```
Comment (japsingh) [inline examples/resource_lacework_integration_azure_agentless_scanning/main.tf:4]: is it really supposed to be lacework/lacework?

File: lacework/resource_lacework_integration_azure_agentless_scanning.go [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+	d.Set("enabled", integration.Enabled == 1)
+	d.Set("integration_level", integration.Data.IntegrationLevel)
+	d.Set("tenant_id", integration.Data.TenantId)
+	d.Set("scanning_subscription_id", integration.Data.ScanningSubscriptionId)
+	d.Set("blob_container_name", integration.Data.BlobContainerName)
+	d.Set("storage_account_url", integration.Data.StorageAccountUrl)
+	d.Set("created_or_updated_time", integration.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", integration.CreatedOrUpdatedBy)
+	d.Set("server_token", integration.ServerToken)
+	d.Set("uri", integration.Uri)
```
Comment (japsingh) [inline lacework/resource_lacework_integration_azure_agentless_scanning.go:386]: resource group name missing here?

File: lacework/resource_lacework_integration_azure_agentless_scanning.go [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+		d.Set("name", integration.Name)
+		d.Set("intg_guid", integration.IntgGuid)
+		d.Set("enabled", integration.Enabled == 1)
+		d.Set("created_or_updated_time", integration.CreatedOrUpdatedTime)
+		d.Set("created_or_updated_by", integration.CreatedOrUpdatedBy)
+		d.Set("integration_level", integration.Type)
+
+		creds := make(map[string]string)
+		creds["client_id"] = integration.Data.Credentials.ClientId
+		d.Set("credentials", []map[string]string{creds})
```
Comment (japsingh) [inline lacework/resource_lacework_integration_azure_agentless_scanning.go:297]: do we need to set client secret?

File: integration/azure_auth.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package integration
+
+import (
+	"encoding/json"
+	"os"
+)
+
+type azureCredentialsFile struct {
+	ClientSecret string `json:"keyId"`
```
Comment (japsingh) [inline integration/azure_auth.go:9]: Are these mapped incorrectly? Shouldn't keyId map to ClientID and secret map to ClientSecret?

**Review by japsingh — APPROVED** [review]
Nice work!
Just a few minor comments.

File: lacework/resource_lacework_integration_azure_agentless_scanning.go (hunk 4 above)
Comment (qianxuweiren) [reply lacework/resource_lacework_integration_azure_agentless_scanning.go:297]: good question. I copied it from the GCP flow. Apparently it was intentionally removed (https://github.com/lacework/terraform-provider-lacework/pull/531) so I'm just going to do the same here.

**Pull request merged**